services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -401,6 +409,7 @@ requestDevice lzusr tid uid = do
     -- device at (almost) the same time.
     provisionLHDevice :: UserId -> Local UserId -> UserLegalHoldStatus -> Sem r ()
     provisionLHDevice zusr luid userLHStatus = do
+      disallowIfMLSUser luid
```

Comment:
maybe a small comment would be nice as well, even though I won't insist on it, the code is already quite clear. 
```suggestion
      -- currently, legalhold is not compatible with MLS, hence we disallow provisioning
      -- a legalhold device for any user who partakes in anything but proteus conversations. 
      disallowIfMLSUser luid
```

Reply:
Took long enough for CI, I think I won't add it. Feel free to create another PR.